### PR TITLE
Kokkos, Spacepoint binning

### PR DESCRIPTION
**Pull request by Yhatoh**
Hi!

I closed the previous PR ( https://github.com/acts-project/traccc/pull/270 ). And in this branch in my fork, i did the changes requested. 

So in the seeding examples, I put the full CPU seeding algorithm and only the spacepoint binning for Kokkos.

Also, I used the format from VS Code to format the code (I think now is correct).

Cheers,
Gabriel

### CODE REVIEW

File: device/kokkos/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+include( traccc-compiler-options-cpp )
+
+traccc_add_library( traccc_kokkos kokkos TYPE SHARED
+  # Utility definitions.
+  "include/traccc/kokkos/utils/definitions.hpp"
+  "include/traccc/kokkos/utils/make_prefix_sum_buff.hpp"
+  "src/utils/make_prefix_sum_buff.cpp"
+  # Seed finding code.
+  "include/traccc/kokkos/seeding/spacepoint_binning.hpp"
+  # CCL code.
```
Comment (krasznaa) [inline device/kokkos/CMakeLists.txt:17]: You should remove this line.

**Review by krasznaa — CHANGES_REQUESTED** [review]
Some changes are needed, but I don't see anything fundamentally wrong here. Just some small cleanup here and there.

**Comment by krasznaa** [comment]
#278 should make this PR build in the CI.

But you still need to fix the formatting of the code. If you have Docker available, the easiest is to do the following:

```
cd traccc/
docker run -it --rm -v $PWD:$PWD -w $PWD --user "$(id -u):$(id -g)" ghcr.io/acts-project/format10:v11
./.github/check_format.sh .
```

And then add all the changes that the [check_format.sh](https://github.com/acts-project/traccc/blob/main/.github/check_format.sh) script makes, to this PR.

**Review by krasznaa — APPROVED** [review]
Okay, let's go ahead with it. So that you can move on to the seed-finding algorithm. :wink:

**Comment by Yhatoh** [comment]
Thanks!

**Pull request merged**